### PR TITLE
shell: Add support to log_panic in RTT backend

### PR DESCRIPTION
**Pull request by nordic-krch**
Shell RTT backend was using RTT call which was using
lock (mutex). In case of panic, that lead to failure
if panic occured in the interrupt context where mutex
is not permitted.

Implementation changed to use write without log in
blocking mode.

Additionally, added static assert to check if raw RTT
log backend is not enabled on channel 0 which is used
by shell RTT backend.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
fyi @chris-makaio @pdunaj

File: subsys/shell/shell_rtt.c [hunk 1]
```diff
@@ -9,6 +9,10 @@
 #include <SEGGER_RTT.h>
 #include <logging/log.h>
 
+BUILD_ASSERT_MSG(!(IS_ENABLED(CONFIG_LOG_BACKEND_RTT) &&
```
Comment (jarz-nordic) [inline subsys/shell/shell_rtt.c]: I was told once by @carlescufi  we should use either: `__ASSERT` or `__ASSERT_NO_MSG` macros

File: subsys/shell/shell_rtt.c [hunk 2]
```diff
@@ -17,6 +21,8 @@ SHELL_DEFINE(shell_rtt, "rtt:~$ ", &shell_transport_rtt,
 
 LOG_MODULE_REGISTER(shell_rtt, CONFIG_SHELL_RTT_LOG_LEVEL);
 
+static bool rtt_blocking;
```
Comment (jarz-nordic) [inline subsys/shell/shell_rtt.c]: shouldn't this be `volatile` ?

File: subsys/shell/shell_rtt.c (hunk 1 above)
Comment (nordic-krch) [reply subsys/shell/shell_rtt.c]: but this is static assert.

Comment (jarz-nordic) [reply subsys/shell/shell_rtt.c]: sorry, missed that !

File: subsys/shell/shell_rtt.c (hunk 2 above)
Comment (nordic-krch) [reply subsys/shell/shell_rtt.c]: i don't think it's needed. It's read only once in `write` function.

**Review by jakub-uC — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**